Examples/minimum_user_for_s2e/src/src_user/TlmCmd/Ccsds/TCFrame.c

```diff
@@ -76,35 +76,46 @@ TCF_VCID TCF_get_vcid(const TCF* tcf)
   }
 }
 
-size_t TCF_get_frame_len(const TCF* tcf)
+size_t TCF_get_frame_len(const TCFrame* tcf)
 {
-  unsigned int pos = 2;
+  uint32_t pos = 2;
   uint8_t mask = 0x03; // 0000 0011b
 
   // pos = 0の下位2bitsとpos = 1の8bitsを合わせた10bits
-  size_t len = (tcf->header[pos] & mask);
+  size_t len = (tcf->packet[pos] & mask);
   len <<= 8;
-  len += tcf->header[pos + 1];
+  len += tcf->packet[pos + 1];
 
   // TC Frameの長さ表記は0起算なので1起算に変換した値を返す
   return len + 1;
 }
 
-uint8_t TCF_get_frame_seq_num(const TCF* tcf)
+uint8_t TCF_get_frame_seq_num(const TCFrame* tcf)
 {
   size_t pos = 4;
-  return (uint8_t)(tcf->header[pos]);
+  return tcf->packet[pos];
 }
 
-uint16_t TCF_get_fecw(const TCF* tcf)
+const TCSegment* TCF_get_tc_segment(const TCFrame* tcf)
 {
-  unsigned int pos = 0;
+  size_t pos = TCF_HEADER_SIZE;
+  return (const TCSegment*)&tcf->packet[pos];
+}
+
+uint16_t TCF_get_fecw(const TCFrame* tcf)
+{
+  size_t length = TCF_get_frame_len(tcf);
 
-  uint16_t fecw = tcf->fecf[pos];
+  uint16_t fecw = tcf->packet[length - 2];
   fecw <<= 8;
-  fecw += tcf->fecf[pos + 1];
+  fecw += tcf->packet[length - 1];
 
   return fecw;
 }
 
+const TCFrame* TCF_convert_raw_byte(const uint8_t* byte)
```

Comment:
[IMO]

```suggestion
const TCFrame* TCF_convert_from_bytes_to_tc_frame(const uint8_t* byte)
```

か


```suggestion
const TCFrame* TCF_deserialize(const uint8_t* byte)
```
とか？

Reply:
上の方で修正しました